libraries/Microsoft.Bot.Builder.AI.QnA/DialogHelper.cs

```diff
@@ -0,0 +1,188 @@
+
+                if (qnaResult != null)
+                {
+                    stepContext.Values[QnAData] = new List<QueryResult>() { qnaResult };
+
+                    var records = new FeedbackRecord[]
+                    {
+                        new FeedbackRecord
+                        {
+                            UserId = stepContext.Context.Activity.Id,
+                            UserQuestion = currentQuery,
+                            QnaId = qnaResult.Id,
+                        }
+                    };
+
+                    var feedbackRecords = new FeedbackRecords { Records = records };
+
+                    // Call Active Learning Train API
+                    await _services.CallTrainAsync(feedbackRecords).ConfigureAwait(false);
+
+                    return await stepContext.NextAsync(new List<QueryResult>() { qnaResult }, cancellationToken).ConfigureAwait(false);
+                }
+                else if (reply.Equals(CardNoMatchText))
+                {
+                    await stepContext.Context.SendActivityAsync(CardNoMatchResponse, cancellationToken: cancellationToken).ConfigureAwait(false);
+                    return await stepContext.EndDialogAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    return await stepContext.ReplaceDialogAsync(ActiveLearningDialogName, stepContext.ActiveDialog.State["options"], cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            return await stepContext.NextAsync(stepContext.Result, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<DialogTurnResult> CheckForMultiTurnPrompt(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            if (stepContext.Result is List<QueryResult> response && response.Count > 0)
+            {
+                // -Check if prompt exists 
+                // -If yes: Add reverse index of prompt display name and its corresponding qna id
+                // -Set PreviousQnAId as response[0].Id
+                // -Set PreviousQuery as stepContext.Values[CurrentQuery]
+                // -Display card for the prompt
+                // -Wait for the reply
+                // -If no: Skip to next step
+            }
+
+            return await stepContext.NextAsync(stepContext.Result, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<DialogTurnResult> DisplayQnAResult(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            if (stepContext.Result is List<QueryResult> response && response.Count > 0)
+            {
+                await stepContext.Context.SendActivityAsync(response[0].Answer, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                var msg = "No QnAMaker answers found.";
```

Comment:
This should be configurable potentially, please consider how that woudld work. Happy to discuss.

Reply:
I have introduced a way to accept NoAnswer text. But we can discuss on better approach.

Reply:
Can;t see it. Did you miss a push?